src/components/live-viewers/live-viewers-manager.tsx

```diff
@@ -46,17 +54,24 @@ export class LiveViewersManager {
 
   private _updateLiveViewers = () => {
     const entryId = this._player.config.sources.id;
-    const ks = this._player.config.session?.ks || null;
     // make api call to get live viewers
-    this._player.provider.doRequest([{loader: GetLiveStatsLoader, params: {ks, entryId}}])
-      .then((data: Map<string, any>) => {
-        if (data && data.has(GetLiveStatsLoader.id)) {
-          const liveStatsLoader = data.get(GetLiveStatsLoader.id);
-          const liveStats = liveStatsLoader?.response;
-          // liveViewers from response is possibly false (in case of 0 viewers)
-          this._liveViewersRef?.current?.updateLiveViewers(liveStats.liveViewers || 0);
-        }
-      });
+    const requestUrl = `${this._baseRequestUrl}/${entryId}`
+    try {
+      fetch(requestUrl)
+        .then(response => response.text())
+        .then((textResponse: any) => {
+          if (textResponse) {
+            const domParser: DOMParser = new DOMParser();
```

Comment:
Can / should we cache the parser and keep a singleton of it ? What's the time cost of creating a parser vs the memory cost of keeping a parser in memory ?

Reply:
irrelevant after latest changes.